packages/hydrogen/src/pagination/Pagination.ts

```diff
@@ -1,12 +1,19 @@
 import {createElement, useEffect, useMemo, useState} from 'react';
 import type {Maybe, PageInfo} from '@shopify/hydrogen/storefront-api-types';
-
+import {flattenConnection} from '@shopify/hydrogen-react';
 import {Link, LinkProps, useNavigation, useLocation} from '@remix-run/react';
 
-type Connection<NodesType> = {
-  nodes: Array<NodesType>;
-  pageInfo: PageInfo;
-};
+type Connection<NodesType> =
+  | {
+      nodes: Array<NodesType>;
+      pageInfo: PageInfo;
+    }
+  | {
+      edges: {
+        nodes: Array<NodesType>;
```

Comment:
```suggestion
        node: Array<NodesType>;
```

For example: https://shopify.dev/docs/api/storefront/2023-04/objects/CollectionEdge